### PR TITLE
DIV-5761 Add test and extra checks for invalid API data

### PR DESCRIPTION
**Pull request by alexwakeman**
[Invalid address information breaks petition](https://tools.hmcts.net/jira/browse/DIV-5761)

### CODE REVIEW

**Comment by alexwakeman** [comment]
This change has been regression tested in PR

File: app/components/AddressLookupStep/addressTypes/postcode.js [hunk 1]
```diff
@@ -96,7 +96,7 @@ module.exports = {
 
       if (ctx.selectAddressIndex !== '-1' && ctx.addresses && ctx.addresses.length) {
         const address = ctx.addresses[ctx.selectAddressIndex];
-        if (address && address.DPA.ADDRESS) {
+        if (address && address.DPA && address.DPA.ADDRESS) {
```
Comment (hosainnet) [inline app/components/AddressLookupStep/addressTypes/postcode.js]: What is the effect visually on the page if address.DPA is not present? i.e do they not get the address populated or do they have to enter it manually?

Comment (alexwakeman) [reply app/components/AddressLookupStep/addressTypes/postcode.js]: We see this if the data is malformed -treated as an error state. Thanks for the review. 

image

File: app/components/AddressLookupStep/addressTypes/postcode.js [hunk 2]
```diff
@@ -73,7 +73,7 @@ module.exports = {
         addresses,
         error
       } = yield postcodeLookupClient.lookupPostcode(ctx.postcode);
-      if (error) {
+      if (error || (Array.isArray(addresses) && addresses.length > 0 && !addresses[0].DPA)) {
```
Comment (dtaylor7) [inline app/components/AddressLookupStep/addressTypes/postcode.js]: move this condition to a variable which explains what it is!

Comment (alexwakeman) [reply app/components/AddressLookupStep/addressTypes/postcode.js]: ok cool done @dtaylor7

**Comment by nsingamsetty** [comment]
This is tested and working fine. Merging now.

**Pull request merged**